### ISSUE TITLE
Regression: Type inference fails on nested zip calls

### ISSUE THREAD

**Issue by pvandyken** [microsoft/pyright#5780]
**Describe the bug**
When nesting a `zip` all inside another `zip` call, type inference fails: the generic argument of the second zip is reported as unknown.

As more of a side note (not sure if this is related to the same issue), it looks to me like the type inference for `bar` in the example below is incorrect: I would expect `zip[tuple[str, ...]]` since we don't know how long the `Iterable` is.

**Code or Screenshots**

Error appears on `pyright>=1.1.314`

```py
# test.py
from __future__ import annotations
from typing import Iterable


def foo(a: Iterable[tuple[str, ...]]):
    bar = zip(*a)
    # zip[tuple[str]]

    _result = zip(a, bar)
    # zip[tuple[tuple[str, ...], tuple[str]]]

    zip(a, zip(*a))
	#      ~~~~~~~
    # Argument type is partially unknown
    #   Argument corresponds to parameter "__iter2" in function "__new__"
    #   Argument type is "zip[tuple[Unknown]]
```

Output:
```
/path/to/test.py
  /path/to/test.py:12:12 - error: Argument type is partially unknown
  Argument corresponds to parameter "__iter2" in function "__new__"
  Argument type is "zip[tuple[Unknown]]
```

**VS Code extension or command-line**
Both

**Comment by erictraut** [comment]
Thanks for the bug report. This is actually caused by the same underlying issue as #5693, which will be fixed in the next release. I've confirmed that the above code no longer produces an error when using the latest "main" branch of pyright.

**Comment by pvandyken** [comment]
Perfect, thanks!

**Comment by erictraut** [comment]
This is included in pyright 1.1.324, which I just published. It will also be included in a future release of pylance.
